source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -141,8 +141,20 @@ void AuthenticatorImpl::startVerify() {
   tokens_.pop_back();
 
   jwt_ = std::make_unique<::google::jwt_verify::Jwt>();
-  ENVOY_LOG(debug, "{}: Parse Jwt {}", name(), curr_token_->token());
-  Status status = jwt_->parseFromString(curr_token_->token());
+  if (provider_) {
+    jwks_data_ = jwks_cache_.findByProvider(provider_.value());
+    cache_jwt_ = jwks_data_->getTokenCache().find(curr_token_->token());
+  }
+  Status status;
+  if (cache_jwt_) {
+    ENVOY_LOG(debug, "{}: Skip parsing, Jwt {} is in Cache.", name(), curr_token_->token());
+    status = Status::Ok;
+    *jwt_ = *cache_jwt_;
```

Comment:
Prefer not to make a copy of jwt here.  It is quite big.

Reply:
Done.